### DISCUSSION TITLE
Android OS (Custom Roms) supporting Google Camera >= 7.6

### DISCUSSION THREAD

**Discussion by lukaspieper** [lukaspieper/Gcam-Services-Provider#28] [General]
Based on the new discoveries from issue #22, I would like to create a list of Android OS (custom roms) that support new Google Camera versions.

You can run this command to get a list of system features your rom supports: `adb shell pm list features`

If the list contains `com.google.android.feature.PIXEL_2017_EXPERIENCE` the rom should "support" at least gcam 7.6.

I assume that the following roms work:

* RattlesnakeOS (only checked [source file](https://github.com/RattlesnakeOS/example-custom-config-repo/blob/11.0/prebuilt/google_experience.xml))
* CalyxOS (only checked [source file](https://gitlab.com/CalyxOS/device_google_redbull/-/blob/android11-qpr1/the_experiences.xml))
* GrapheneOS (tested, [release notes](https://grapheneos.org/releases#2021.06.20.20))

Thanks.

**Comment by innir** [comment]
It's included in the current GrapheneOS beta release as well :-) See e.g. https://github.com/GrapheneOS/device_google_sunfish/commit/faff8147f65e7b69dc26ce86a3e51a55ef0252ac

**Reply by lukaspieper** [reply]
As a GrapheneOS user myself, thank you for the contribution. I have added it to the list and can confirm that Gcam 8.2 works for me (Pixel 3).

**Reply by tifriis208** [reply]
A great thanks ! I'm also a GrapheneOS user wither Pixel 3. DO you know if there is a tutorial to install 8.2 (I'm not familiar with "--force-queryable"). WHere did you get the apk ? Thanks for your help :)

**Reply by lukaspieper** [reply]
There is no _tutorial_ as far as I know. Issue #6 includes multiple ways to install the app with `--force-queryable`. However all of them are quite technical.

**Comment by chenfeng858** [comment]
Sir, I am currently using the LineageOS 23 (A16) ROM and it seems to have a problem that does not exist in 22.2. When I used LineageOS 22.2 (A15) before, I installed your software LMC.8.4 and it could be opened directly, but after 23, I need to open the front camera and then the rear camera to see the picture. Can you optimize it for the LineageOS 23 version?

**Reply by lukaspieper** [reply]
Hi,

please create a issue and select "Bug report", then provide all the information the template asks you to provide.

Thanks.

**Comment by Danny3** [comment]
I can confirm it works with LineageOS 23.2 (Android 16) too, on a Samsung Galaxy S10+ (Exynos) phone, with:
Shamim 8.8: [SGCAM_8.8.224.10_STABLE_V19]
From:
https://www.celsoazevedo.com/files/android/google-camera/dev-shamim/
The:
Shamim 9.1: [SGCAM_9.1.098.24_STABLE_V10] doesn't work as it makes the whole OS crash and forces a restart of the phone.
And the Gcam apps from other developers just crash themselves.
